flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/UnsignedTypeConversionITCase.java

```diff
@@ -0,0 +1,214 @@
+ * Test unsigned type conversion between Flink and JDBC driver mysql, the test underlying use
+ * MariaDB to mock a DB which use mysql driver too.
+ */
+public class UnsignedTypeConversionITCase {
+
+	private static final String DEFAULT_DB_NAME = "test";
+	private static final String TABLE_NAME = "unsigned_test";
+	private static final String[] DATA = new String[]{
+		"127",
+		"255",
+		"32767",
+		"65535",
+		"2147483647",
+		"4294967295",
+		"9223372036854775807",
+		"18446744073709551615"};
+
+	private StreamTableEnvironment tEnv;
+	private String dbUrl;
+	private Connection connection;
+
+	@ClassRule
+	public static MariaDB4jRule db4jRule = new MariaDB4jRule(
+		DBConfigurationBuilder.newBuilder().setPort(3306).build(),
+		DEFAULT_DB_NAME,
+		null);
+
+	@Before
+	public void setUp() throws SQLException, ClassNotFoundException {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		tEnv = StreamTableEnvironment.create(env);
+		//dbUrl: jdbc:mysql://localhost:3306/test
+		dbUrl = db4jRule.getURL();
+		connection = DriverManager.getConnection(dbUrl);
+		// create mysql table
+		PreparedStatement ddlStatement = connection.prepareStatement("create table " + TABLE_NAME + " (" +
+			" tiny_c TINYINT," +
+			" tiny_un_c TINYINT UNSIGNED," +
+			" small_c SMALLINT," +
+			" small_un_c SMALLINT UNSIGNED," +
+			" int_c INTEGER ," +
+			" int_un_c INTEGER UNSIGNED," +
+			" big_c BIGINT," +
+			" big_un_c BIGINT UNSIGNED);");
+		ddlStatement.execute();
+
+		// create flink table
+		String sourceDDL = "create table jdbc_table (" +
+			"tiny_c TINYINT," +
+			"tiny_un_c SMALLINT," +
+			"small_c SMALLINT," +
+			"small_un_c INT," +
+			"int_c INT," +
+			"int_un_c BIGINT," +
+			"big_c BIGINT," +
+			"big_un_c DECIMAL(20, 0)) with(" +
+			" 'connector' = 'jdbc'," +
+			" 'url' = '" + dbUrl + "'," +
+			" 'table-name' = '" + TABLE_NAME + "'" +
+			")";
+		tEnv.executeSql(sourceDDL);
+	}
+
+	@Test
+	public void testReadUnsignedType() throws SQLException {
+		// insert data to db
+		String insertSql = String.format("insert into " + TABLE_NAME +
+			"(tiny_c, tiny_un_c, small_c, small_un_c, int_c, int_un_c, big_c, big_un_c) values (" +
+			"%s, %s, %s, %s, %s, %s, %s, %s)", DATA);
+		PreparedStatement insertStatement = connection.prepareStatement(insertSql);
+		insertStatement.execute();
+		// read data from db
+		Iterator<Row> res = tEnv.executeSql("select tiny_c, tiny_un_c, small_c, small_un_c," +
+			" int_c, int_un_c, big_c, big_un_c from jdbc_table")
+			.collect();
+		List<Row> actual = new ArrayList<>();
+		while (res.hasNext()) {
+			actual.add(res.next());
+		}
+		assertTrue(actual.size() == 1);
+		assertTrue(actual.get(0).getArity() == DATA.length);
+		String[] expected = new String[DATA.length];
+		for (int i = 0; i < DATA.length; i++) {
+			expected[i] = actual.get(0).getField(i).toString();
+		}
+		assertArrayEquals(DATA, expected);
+	}
+
+	@Test
+	public void testWriteUnsignedType() throws Exception {
```

Comment:
Could you combine these two tests into one? We can test writing values into the table, and verify the data using jdbc connection. And then test reading values from the table.